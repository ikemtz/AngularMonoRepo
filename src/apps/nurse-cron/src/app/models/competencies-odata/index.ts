/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './competency.form-group-fac';
export * from './competency.form';
export * from './competency.model';
export * from './competency.properties';
export * from './competency.test-obj-fac';
