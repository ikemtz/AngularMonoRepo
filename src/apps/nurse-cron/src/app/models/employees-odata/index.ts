/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './employee-certification.form-group-fac';
export * from './employee-certification.form';
export * from './employee-certification.model';
export * from './employee-certification.properties';
export * from './employee-certification.test-obj-fac';
export * from './employee-competency.form-group-fac';
export * from './employee-competency.form';
export * from './employee-competency.model';
export * from './employee-competency.properties';
export * from './employee-competency.test-obj-fac';
export * from './employee-health-item.form-group-fac';
export * from './employee-health-item.form';
export * from './employee-health-item.model';
export * from './employee-health-item.properties';
export * from './employee-health-item.test-obj-fac';
export * from './employee.form-group-fac';
export * from './employee.form';
export * from './employee.model';
export * from './employee.properties';
export * from './employee.test-obj-fac';
