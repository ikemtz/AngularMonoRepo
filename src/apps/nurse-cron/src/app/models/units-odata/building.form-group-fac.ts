/* eslint-disable @typescript-eslint/no-unused-vars */
/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR
import { IBuildingForm } from './building.form';

export function BuildingFormGroupFac(): FormGroup<IBuildingForm> {
  return new FormGroup<IBuildingForm>({
    id: new FormControl<string | null | undefined>(null),
    name: new FormControl<string>('', {
      validators: Validators.compose([
        Validators.required,
        Validators.minLength(1),
        Validators.maxLength(50),
      ]),
      nonNullable: true,
    }),
    siteName: new FormControl<string>('', {
      validators: Validators.compose([
        Validators.required,
        Validators.minLength(1),
        Validators.maxLength(150),
      ]),
      nonNullable: true,
    }),
    addressLine1: new FormControl<string>('', {
      validators: Validators.compose([
        Validators.required,
        Validators.minLength(1),
        Validators.maxLength(250),
      ]),
      nonNullable: true,
    }),
    addressLine2: new FormControl<string | null | undefined>(null, {
      validators: Validators.maxLength(250),
    }),
    cityOrMunicipality: new FormControl<string>('', {
      validators: Validators.compose([
        Validators.required,
        Validators.minLength(1),
        Validators.maxLength(250),
      ]),
      nonNullable: true,
    }),
    stateOrProvidence: new FormControl<string>('', {
      validators: Validators.compose([
        Validators.required,
        Validators.minLength(1),
        Validators.maxLength(50),
      ]),
      nonNullable: true,
    }),
    postalCode: new FormControl<string>('', {
      validators: Validators.compose([
        Validators.required,
        Validators.minLength(1),
        Validators.maxLength(50),
      ]),
      nonNullable: true,
    }),
    country: new FormControl<string | null | undefined>(null, {
      validators: Validators.maxLength(3),
    }),
    gpsData: new FormControl<string | null | undefined>(null),
    deletedBy: new FormControl<string | null | undefined>(null, {
      validators: Validators.maxLength(250),
    }),
    deletedOnUtc: new FormControl<Date | null | undefined>(null),
  });
}
