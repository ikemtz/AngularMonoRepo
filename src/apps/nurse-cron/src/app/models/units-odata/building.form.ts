/* eslint-disable @typescript-eslint/no-unused-vars */
/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #form.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormArray, FormGroup } from '@angular/forms'; //NOSONAR

export interface IBuildingForm {
  id?: FormControl<string | null>;
  name?: FormControl<string>;
  siteName?: FormControl<string>;
  addressLine1?: FormControl<string>;
  addressLine2?: FormControl<string | null>;
  cityOrMunicipality?: FormControl<string>;
  stateOrProvidence?: FormControl<string>;
  postalCode?: FormControl<string>;
  country?: FormControl<string | null>;
  gpsData?: FormControl<string | null>;
  deletedBy?: FormControl<string | null>;
  deletedOnUtc?: FormControl<Date | null>;
}
