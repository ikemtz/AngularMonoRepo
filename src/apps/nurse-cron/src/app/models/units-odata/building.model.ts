/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface IBuilding {
  id?: string;
  name?: string;
  siteName?: string;
  addressLine1?: string;
  addressLine2?: string;
  cityOrMunicipality?: string;
  stateOrProvidence?: string;
  postalCode?: string;
  country?: string;
  gpsData?: string;
  deletedBy?: string;
  deletedOnUtc?: Date;
}
