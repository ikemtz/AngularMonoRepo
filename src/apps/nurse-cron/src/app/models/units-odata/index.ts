/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './building.form-group-fac';
export * from './building.form';
export * from './building.model';
export * from './building.properties';
export * from './building.test-obj-fac';
export * from './unit.form-group-fac';
export * from './unit.form';
export * from './unit.model';
export * from './unit.properties';
export * from './unit.test-obj-fac';
