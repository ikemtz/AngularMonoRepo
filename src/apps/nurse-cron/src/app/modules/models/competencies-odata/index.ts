/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './competency.form-group-fac';
export * from './competency.model';
export * from './competency.properties';
