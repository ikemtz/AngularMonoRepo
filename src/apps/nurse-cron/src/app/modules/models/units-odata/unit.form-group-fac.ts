/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function UnitFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    buildingId: new FormControl('', Validators.required),
    name: new FormControl('', Validators.compose([Validators.required, Validators.maxLength(102)])),
    del: new FormControl(''),
    roomCount: new FormControl('', Validators.required),
    deletedBy: new FormControl('', Validators.maxLength(250)),
    deletedOnUtc: new FormControl(''),
    building: new FormControl(''),
  });
}
