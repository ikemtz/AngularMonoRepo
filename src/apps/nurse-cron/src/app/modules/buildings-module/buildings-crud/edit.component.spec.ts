import { CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA } from '@angular/core';
import { waitForAsync, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { DatePickerModule } from '@progress/kendo-angular-dateinputs';
import { createDataEntryMockFacade } from 'imng-kendo-data-entry/testing';
import { of } from 'rxjs';
import {
  mockConsoleError,
  mockConsoleGroup,
  mockConsoleWarn,
} from 'imng-ngrx-utils/testing';

import { BuildingEditComponent } from './edit.component';
import { BuildingCrudFacade } from './crud.facade';
import { BuildingProperties, IBuilding } from '../../../models/units-odata';

describe('BuildingEditComponent', () => {
  let component: BuildingEditComponent;
  let fixture: ComponentFixture<BuildingEditComponent>;
  let facade: BuildingCrudFacade;
  let consoleWarnMock: jest.SpyInstance<void>;
  let consoleGroupMock: jest.SpyInstance<void>;

  beforeEach(waitForAsync(() => {
    consoleWarnMock = mockConsoleWarn();
    consoleGroupMock = mockConsoleGroup();
    TestBed.configureTestingModule({
      declarations: [BuildingEditComponent],
      imports: [ReactiveFormsModule, NoopAnimationsModule, DatePickerModule],
      providers: [
        {
          provide: BuildingCrudFacade,
          useValue: createDataEntryMockFacade({ currentEntity$: of({}) }),
        },
      ],
      schemas: [CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(BuildingEditComponent);
    component = fixture.componentInstance;
    facade = TestBed.inject(BuildingCrudFacade);
    fixture.detectChanges();
  });

  afterAll(() => {
    component.ngOnDestroy();
    consoleWarnMock.mockRestore();
    consoleGroupMock.mockRestore();
  });

  test('should update', () => {
    component.initForm();
    component.addEditForm.patchValue({
      [BuildingProperties.ID]: 'ID',
      [BuildingProperties.NAME]: 'NAME',
      [BuildingProperties.SITE_NAME]: 'SITE_NAME',
      [BuildingProperties.ADDRESS_LINE_1]: 'ADDRESS_LINE_1',
      [BuildingProperties.ADDRESS_LINE_2]: 'ADDRESS_LINE_2',
      [BuildingProperties.CITY_OR_MUNICIPALITY]: 'CITY_OR_MUNICIPALITY',
      [BuildingProperties.STATE_OR_PROVIDENCE]: 'STATE_OR_PROVIDENCE',
      [BuildingProperties.POSTAL_CODE]: 'POSTAL_CODE',
      [BuildingProperties.COUNTRY]: 'COU',
      [BuildingProperties.GPS_DATA]: 'GPS_DATA',
      [BuildingProperties.DELETED_BY]: 'DELETED_BY',
      [BuildingProperties.DELETED_ON_UTC]: new Date(),
    });
    let item: IBuilding | undefined;
    facade.updateExistingEntity = jest.fn((x) => (item = x));
    expect(component.getFormErrors()).toStrictEqual([]);
    component.onSubmit();
    expect(facade.saveNewEntity).toHaveBeenCalledTimes(0);
    expect(facade.updateExistingEntity).toHaveBeenCalledTimes(1);

    expect(item).toMatchSnapshot({
      deletedOnUtc: expect.any(Date),
    });
  });

  /**
   * Note: if this test fails, then you're missing validators in your forms.
   * Using form validators is typically a good idea.
   */
  test('should not update', () => {
    const consoleErrorMock = mockConsoleError();
    component.addEditForm?.patchValue({});
    component.onSubmit();
    expect(facade.saveNewEntity).toHaveBeenCalledTimes(0);
    expect(facade.updateExistingEntity).toHaveBeenCalledTimes(0);
    consoleErrorMock.mockRestore();
  });

  test('should cancel', () => {
    component.cancel();
    expect(facade.clearCurrentEntity).toHaveBeenCalledTimes(1);
  });
});
