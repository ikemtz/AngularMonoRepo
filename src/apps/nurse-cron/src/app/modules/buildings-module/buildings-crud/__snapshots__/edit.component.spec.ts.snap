// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildingEditComponent should update 1`] = `
{
  "addressLine1": "ADDRESS_LINE_1",
  "addressLine2": "ADDRESS_LINE_2",
  "cityOrMunicipality": "CITY_OR_MUNICIPALITY",
  "country": "COU",
  "deletedBy": "DELETED_BY",
  "deletedOnUtc": Any<Date>,
  "gpsData": "GPS_DATA",
  "id": "ID",
  "name": "NAME",
  "postalCode": "POSTAL_CODE",
  "siteName": "SITE_NAME",
  "stateOrProvidence": "STATE_OR_PROVIDENCE",
}
`;
